examples/Mixtures/slot_based.py

```diff
@@ -0,0 +1,258 @@
+# to your mixture, the permutation invariance constraint should be discarded and one
+# could further argue that adding the same substance multiple times should be allowed.
+# ```
+
+#### Duplicate Substances
+
+# Assuming that the order of addition is irrelevant, there is no difference between
+# having two slots with the same substance or having only one slot with the combined
+# amounts. Thus, we want to make sure that there are no such duplicate label entries:
+
+no_duplicates_constraint = DiscreteNoLabelDuplicatesConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"]
+)
+
+#### Permutation Invariance
+
+# Next, we need to take care of permutation invariance. If our order of addition does
+# not matter, the result of interchanging any two slots does not alter the overall
+# mixture, i.e. the mixture slots are are considered permutation-invariant.
+
+# A complication with permutation invariance arises from the fact that we have not
+# only a label per slot, but also a numerical amount. If this amount is zero, then the
+# label of the slot becomes meaningless (i.e. which ingredient should be considered for the
+# slot), because adding zero of it does not change the mixture. In BayBE, we call
+# this a "dependency", i.e. the slot labels depend on the slot amounts and are only
+# relevant if the amount satisfies some condition (in this case "amount > 0").
+
+# The {class}`~baybe.constraints.discrete.DiscreteDependenciesConstraint` informs the
+# {class}`~baybe.constraints.discrete.DiscretePermutationInvarianceConstraint` about
+# these dependencies so that they are correctly included in the filtering process:
+
+perm_inv_constraint = DiscretePermutationInvarianceConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"],
+    dependencies=DiscreteDependenciesConstraint(
+        parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+        conditions=[
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+        ],
+        affected_parameters=[["Slot1_Label"], ["Slot2_Label"], ["Slot3_Label"]],
+    ),
+)
+
+#### Substance Amounts
+
+# Interpreting the slot amounts as percentages, we need to ensure that their total is
+# always 100:
+
+sum_constraint = DiscreteSumConstraint(
+    parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+    condition=ThresholdCondition(threshold=100, operator="=", tolerance=SUM_TOLERANCE),
+)
+
+
+# We store all constraints in a single list:
+
+constraints = [perm_inv_constraint, sum_constraint, no_duplicates_constraint]
+
+
+### Search Space Creation
+
+# With all building blocks in place, we can now assemble our discrete space and inspect
+# its configurations:
+
+space = SubspaceDiscrete.from_product(parameters=parameters, constraints=constraints)
+print(space.exp_rep)
```

Comment:
@AVHopp: I wanted to give the `to_markdown` version another try here to get some nice rendering. But the the file conversion fails when I add it (`tabulate` needs to be installed, but I did that). Can you have a look?

Reply:
I guess you are referring to the pandas function? Did you try to just use the already existing `pretty_print` function?

Reply:
I was hoping more to get an actual rendered table, but we can also try with `pretty_print` first 👍🏼

Reply:
I think the result is OK 👍🏼